app/server/utils/ussd/token_processor.py

```diff
@@ -61,75 +64,92 @@ def transfer_token(sender: User, recipient: User, amount: float, reason_id: Opti
         sent_token = default_token(sender)
         received_token = default_token(recipient)
 
+        transfer_use = None
+        if reason_id is not None:
+            transfer_use = str(int(reason_id))
         transfer = make_payment_transfer(amount, token=sent_token, send_user=sender, receive_user=recipient,
-                                         transfer_use=reason_id)
-        received_amount = amount
+                                         transfer_use=transfer_use, is_ghost_transfer=True,
+                                         require_sender_approved=False, require_recipient_approved=False)
+        exchanged_amount = None
 
         if sent_token.id != received_token.id:
             exchange = Exchange()
             exchange.exchange_from_amount(user=recipient, from_token=sent_token, to_token=received_token,
                                           from_amount=amount, dependent_task_ids=[transfer.blockchain_task_id])
-            received_amount = exchange.to_transfer.to_amount
+            exchanged_amount = exchange.to_transfer.transfer_amount
 
-        return received_amount
+        return exchanged_amount
 
     @staticmethod
     def send_balance_sms(user: User):
         def get_token_info(transfer_account: TransferAccount):
             token = transfer_account.token
             limit = TokenProcessor.get_limit(user, token)
+            exchange_rate = TokenProcessor.get_exchange_rate(user, token)
             return {
-                # TODO: is it name or symbol?
-                "name": token.name,
+                "name": token.symbol,
                 "balance": transfer_account.balance,
                 "limit": limit.transfer_balance_percentage,
-                "exchange_rate": TokenProcessor.get_exchange_rate(user, token)
+                "exchange_rate": str(round(exchange_rate, 2))
             }
-        # TODO: should we merge transfer accounts with same tokens..? any currencies we should filter out?
-        token_info = map(get_token_info, user.transfer_accounts)
-        token_balances = "\n".join(map(lambda x: f"{x['name']} {x['balance']}", token_info))
-        # this one is also not well generalized currently - how get real world currency name of reserve token?
-        token_exchanges = "\n".join(
-            map(lambda x: f"{x['limit'] * x['balance']} {x['name']} (1 {x['name']} = {x['exchange_rate']} KSH)",
-                token_info))
+
+        reserve_token = user.get_reserve_token()
+        # transfer accounts could be created for other currencies exchanged with, but we don't want to list those
+        # could there be multiple reserve? should we filter out all reserve tokens?
+        transfer_accounts = filter(lambda x: x.is_ghost is not True and x.token_id != reserve_token.id,
```

Comment:
currently if a user has any non-liquid tokens aside from the master reserve, `get_limit` will error out. might be a bit aggressive to error out, but we need to ensure we don't try to do the limit stuff on non-grassroots tokens

Reply:
So we need a 'GE' flag or something to ensure the limit doesn't get applied to non-ge tokens?

Reply:
or i could drop the raise exception and make it so we get the first limit that has a percentage on it. but then if we add other percentage based limits in the future and they accidentally get applied to GE users, it could produce the wrong behavior

Reply:
> but then if we add other percentage based limits in the future and they accidentally get applied to GE users, it could produce the wrong behavior

That feels bad. My instinct is that filtering should just be smart enough to filter as required. 'Explicit is better than implicit' as well...